nixos/modules/services/editors/infinoted.nix

```diff
@@ -124,33 +125,34 @@ in {
 
         wantedBy = [ "multi-user.target" ];
         after = [ "network.target" ];
-        path = [ cfg.package ];
 
         serviceConfig = {
           Type = "simple";
           Restart = "always";
           ExecStart = "${cfg.package}/bin/infinoted-0.6 --config-file=/var/lib/infinoted/infinoted.conf";
+          User = cfg.user;
+          Group = cfg.group;
+          ExecStartPre = "+" + pkgs.writeScript "infinoted-setup" ''
```

Comment:
I believe  you'll need `PermissionsStartOnly` for this to work, otherwise it'll run as `User`. Also, I suggest going back to using `preScript`, it should expand to something more or less like this, just neater to read.